pyiron_atomistics/lammps/interactive.py

```diff
@@ -242,6 +243,84 @@ def interactive_initialize_interface(self):
         self._reset_interactive_run_command()
         self.interactive_structure_setter(self.structure)
 
+    def set_callback(self, function, n_call=1, n_apply=1, overload_internal_callback=False):
+        """
+        **********************
+        *** Expert feature ***
+        **********************
+
+        Set callback function that modifies forces.
+
+        Args:
+            function (function): User-defined function that returns forces (see below)
+            n_call (int): Make callback every `n_call` steps
+            n_apply (int): Apply callback forces every `n_apply` steps
+            overload_internal_callback (bool): Whether to overload internal callback (see below).
+                Overloading the internal callback function will have the advantage that the code
+                will not need to do expensive copying, BUT it is extremely error-prone. Make
+                sure that the code works without overloading the internal callback function first.
+
+        `function` must have the following form:
+
+        ```
+        def function(positions, ntimestep, nlocal):
+            your_evaluation
+            return forces
+        ```
+        where `positions` is the positions of all atoms on the local processor, `ntimestep` is the
+        current timestep and `nlocal` is the number of atoms on the current processor. `forces`
+        must be of the shape `(n_atoms, 3)`. The total translational force will be eliminated
+        inside pyiron.
+
+        If `overload_internal_callback` is set to `True`, then `function` must have the following
+        form:
+
+        ```
+        def function(ptr, timestep, nlocal, ids, x, fexternal):
+            your_evaluation
+        ```
+
+        with the following arguemnts:
+
+        - `ptr`: pointer provided by and simply passed back to external driver
+        - `timestep`: current LAMMPS timestep
+        - `nlocal`: # of atoms on this processor
+        - `ids`: list of atom IDs on this processor
+        - `x`: coordinates of atoms on this processor
+        - `fexternal`: forces to add to atoms on this processor
+
+        Note: Do NOT overwrite `fexternal`, because it points to the internal memory of LAMMPS and
+        therefore overwriting it will erase its functionality. E.g. DO `fexternal.fill(0)` and NOT
+        `fexternal = np.zeros_like(x)`.
+
+        Example: Add random forces
+
+        ```
+        from pyiron import Project
+
+        def random_forces(x, *args):
+            return 0.1 * np.random.randn(*x.shape)
+
+        pr = Project('RANDOM')
+        lmp = pr.create.job.Lammps('random_forces')
+        lmp.structure = your_structure
+        lmp.potential = your_potential
+        lmp.server.run_mode.interactive = True
```

Comment:
```suggestion
        lmp.interactive_open()
```

For symmetry with the `interactive_close()` at the end.

You could also consider delaying this until [base 565](https://github.com/pyiron/pyiron_base/pull/565) is finished and merged so that you can just use `with lmp.interactive_open() as lmp_int:` and get rid of the close entirely, but if you'd rather merge this sooner that's cool too.

Reply:
Ok since it's just about DocString and the functionality will anyway not change, I keep the current one (i.e. with `interactive_open/_close`), but I'd use `with` if it happens to be already ready by the time this PR is to be merged.